csrc/multidevice/executor.cpp

```diff
@@ -86,24 +86,24 @@ std::unordered_map<Val*, c10::IValue> MultiDeviceExecutor::allocateRecvBuffers(
   if (fusion_copy->outputs().empty()) {
     return {};
   }
-  // TODO: Not working with FusionExecutorCache
-  FusionExecutor fe;
-  fe.compileFusion(fusion_copy.get(), global_inputs_IValues);
-  auto buffers = fe.allocOutputSpace(global_inputs_IValues);
+
+  FusionExecutorCache fec(std::move(fusion_copy), 0, false);
```

Comment:
Can we also save this fec in the executor so it is only compiled once?

Reply:
sure, good idea

Reply:
Actually the best would be to simply cache the necessary allocations sizes, this is even more lightweight. I will do it on a separate PR.

Note that in the (hopefully near :) ) future we will melt `MultiDeviceExecutor` into the main path of `FusionExecutorCache`. Then, the allocation will be taken care of automatically there, and so, in my opinion, let's not worry to much about the allocation flow we have here.